### PR TITLE
DOC: Improve doc of KL functionals, see #743

### PR DESCRIPTION
**Pull request by adler-j**
Improved the math documentation of the KL functionals to be more readable to the "layman".

### CODE REVIEW

File: odl/solvers/functional/default_functionals.py [hunk 1]
```diff
@@ -1217,18 +1223,23 @@ class KullbackLeiblerCrossEntropy(Functional):
...
-    For further information about the functional, see for example `this article
+        F(x)
+        =
+        \\begin{cases}
+            \\sum_{i} \left( g_i - x_i + x_i \log \left( \\frac{x_i}{g_i}
+            \\right) \\right)
+            & \\text{if } x_i > 0 \\forall i
+            \\\\
+            +\\infty & \\text{else}
+        \\end{cases}
```
Comment (kohr-h) [inline odl/solvers/functional/default_functionals.py]: I'm confused. This definition is the same as for the usual KL divergence, which is correct, but shouldn't one of them be a function of `x` and the other one a function of `g`?

Comment (adler-j) [inline odl/solvers/functional/default_functionals.py]: I think this is a naming issue. Paging @aringh who wrote the original documentation. I simply ported the old doc to more readable mathematics (and added what I presumed was the free variable, given that we had "where g is the prior" before.)

Comment (kohr-h) [inline odl/solvers/functional/default_functionals.py]: I assume that this one should be `F(g)` instead for fixed `x`. Or you switch roles completely.

Comment (aringh) [inline odl/solvers/functional/default_functionals.py]: The definition is the same, up to the fact that we switch which of the two arguments is seen as prior (i.e., is fixed) and which is seen as variable (i.e., the one we optimize over).

If you see it as a functional of two arguments
```
F(x, y) = \\begin{cases}
             \\sum_{i} \left( y_i - x_i + x_i \log \left( \\frac{x_i}{y_i}
             \\right) \\right)
             & \\text{if } x_i > 0 \\forall i \text{and } y_i > 0 \\forall i
             \\\\
             +\\infty & \\text{else}
         \\end{cases}
```
the one of the functionals is given by fixing `x`, the other by fixing `y`.

Comment (adler-j) [inline odl/solvers/functional/default_functionals.py]: So then the conclusion is that this text is correct?

Comment (adler-j) [inline odl/solvers/functional/default_functionals.py]: Above comment is "clearly" wrong (my comment, that is). Anyway, this is then a function of `g` I suppose? What would we call `x` in that case?

**Comment by adler-j** [comment]
Updated according to comments and added reference. Ok now? Need a review from @aringh I think.

File: odl/solvers/functional/default_functionals.py (hunk 1 above)
Comment (aringh) [inline odl/solvers/functional/default_functionals.py]: No, I think what you had was correct. The naming in the two functionals where different: `x` is always the variable and `g` is always the prior. As it says now, when `g` is the variable in ` KullbackLeiblerCrossEntropy`, the two functionals are the same.

File: odl/solvers/functional/default_functionals.py [hunk 2]
```diff
@@ -1217,30 +1229,41 @@ class KullbackLeiblerCrossEntropy(Functional):
...
-        \\sum_{i} \left( g_i - x_i + x_i \log \left( \\frac{x_i}{ pos(g_i) }
-        \\right) \\right) + I_{x \\geq 0}(x)
-
-    where :math:`g` is the prior, and :math:`I_{x \\geq 0}(x)` is the indicator
-    function on nonnegative elements.
-
-    `Wikipedia article on Kullback Leibler divergence
-    <https://en.wikipedia.org/wiki/Kullback%E2%80%93Leibler_divergence>`_.
-    For further information about the functional, see for example `this article
+        F(g)
```
Comment (aringh) [inline odl/solvers/functional/default_functionals.py]: Just change to a function of `x`: `F(x)`.

File: odl/solvers/functional/default_functionals.py [hunk 3]
```diff
@@ -1356,12 +1379,11 @@ class KullbackLeiblerCrossEntropyConvexConj(Functional):
 
     Notes
     -----
-    The functional is given by
+    The functional :math:`F^*` with prior :math:`x>0` is given by
 
     .. math::
-        \\sum_i g_i (exp(x_i) - 1)
+        F^*(g) = \\sum_i g_i \\left(e^{x_i} - 1\\right)
```
Comment (aringh) [inline odl/solvers/functional/default_functionals.py]: Just change to a function of `x`: `F(x)`.

**Comment by adler-j** [comment]
> I also saw when looking at the code that the convex conjugate functionals are not exposed in all at the top. Do we want them to be that?

Leaving them as is for the moment, the interested user can fish them out. I have yet to find a use-case for them.

**Comment by adler-j** [comment]
Merge after CI

**Pull request merged**